components/03-modules/form/_form.scss

```diff
@@ -24,21 +25,15 @@
 
             &__field {
                 width: 100%;
-                border: $button-icon-border;
                 line-height: $button-height;
                 height: $button-height;
-                border-radius: 0;
-                @include mq($screen-s) {
-                    width: 290px;
-                }
-            }
-
-            &__label {
-                margin: 0 10px 0 0;
+                border-top-right-radius: 0;
+                border-bottom-right-radius: 0;
             }
         }
 
         .button {
+            white-space: nowrap;
```

Comment:
Buttons need to have flexible height. Probably replacing `width` to `min-width` do the job and you will be able to remove this line.